uber/templates/macros.html

```diff
@@ -0,0 +1,115 @@
+{% macro ie7_compatibility_check() -%}
+    <!--[if lte IE 7]>
+        <script>
+            if (!document.documentMode) { //if documentMode exists, this is a later IE in compatibility mode
+                alert("This website requires IE 8 or later.");
+            }
+        </script>
+    <![endif]-->
+{%- endmacro %}
+
+
+{% macro stripe_button(label) -%}
+    <button class="stripe-button-el">
+        <span class="display: block; min-height: 30px;">{{label}}</span>
+    </button>
+{%- endmacro %}
+
+
+{% macro popup_link(href, text='<sup>?</sup>') -%}
+    <a onClick="window.open('{{ href }}', 'info', 'toolbar=no,height=500,width=375,scrollbars=yes').focus(); return false;"
+       href="{{ href }}">{{ text|safe }}</a>
+
+    {#
+    TODO: integrate these changes from upstream, if needed. they may not be needed anymore
+
+    inner_text = "window.open('{self.href}', 'info', 'toolbar=no,height=500,width=375,scrollbars=yes').focus(); return false;".format(self=self)
+    inner_text = inner_text.replace("'", "&quot;")
+    return "<a onClick='{inner_text}' href='{self.href}'>{self.text}</a>".format(inner_text=inner_text, self=self)
+    #}
+{%- endmacro %}
+
+
+{% macro checkbox(model, field_name, id=None) -%}
+    <input
+        type="checkbox"
+        name="{{ field_name }}"
+        value="1"
+        {% if id %}id="{{ id }}"{% endif %}
+        {% if model[field_name] %}checked{% endif %} />
+{%- endmacro %}
+
+
+{% macro checkgroup(model, field_name, no_instance=False) -%}
+    {% set defaults = model[field_name] if not no_instance else None %}
+    {% set defaults = defaults.split(",") if defaults else [] %}
+
+    {% for num, desc in model.get_field(field_name).type.choices -%}
+        <label style="font-weight: normal;">
+            <input
+                type="checkbox"
+                name="{{ field_name }}"
+                value="{{ num }}"
+                {% if num|string in defaults %}checked{% endif %} />
+           {{ desc }}
+        </label>
+        {% if not loop.last %}&nbsp;&nbsp;{% endif %}
+    {%- endfor %}
+{%- endmacro %}
+
+
+{% macro radiogroup(model, field_name, onchanged="donationChanged();") -%}
+    {# TODO: remove hardcoded "donationChanged()" param #}
```

Comment:
probably implement the suggestion of this TODO regarding the ```onchanged``` param

this is likely a comment I left to myself to catch in the final code review